apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -308,6 +309,20 @@ impl Plugin for Telemetry {
     }
 }
 
+impl Drop for Telemetry {
+    fn drop(&mut self) {
+        // This allows us to make sure the telemetry related globals
```

Comment:
Gut feeling is that this will activate even if it was not supposed to, which would make this incorrect.
Let's talk this through.